ui/StatusQ/src/submodelproxymodel.cpp

```diff
@@ -126,28 +171,128 @@ void SubmodelProxyModel::setSubmodelRoleName(const QString& sumodelRoleName)
     initializeIfReady();
 }
 
+void SubmodelProxyModel::onCustomRoleChanged(QObject* source, int role)
+{
+    if (!m_dataChangedQueued) {
+        m_dataChangedQueued = true;
+        QMetaObject::invokeMethod(this, "emitAllDataChanged", Qt::QueuedConnection);
+    }
+}
+
+void SubmodelProxyModel::emitAllDataChanged()
+{
+    m_dataChangedQueued = false;
+    auto count = rowCount();
+
+    if (count == 0)
+        return;
+
+    QVector<int> roles;
```

Comment:
```suggestion
    const auto roles = m_additionalRolesMap.values();
```

and remove the `for` loop below :)

Reply:
Yeah, you could convert it with to a QVector with `{list.cbegin(), list.cend()}` but nvm, not a big deal

Reply:
Ahh, it can be done even this way:
```
    QVector<int> roles(m_additionalRolesMap.cbegin(), m_additionalRolesMap.cend());
```
Fixed, thx!

Reply:
Yeah, that's what I had in my mind:
```cpp
emit this->dataChanged(index(0, 0), index(count - 1, 0), {m_additionalRolesMap.cbegin(), m_additionalRolesMap.cend()});
```
but didn't realize the vectorvs. list difference since here you don't even have to declare it